API/projectClass/layersClassRealSLD.m

```diff
@@ -0,0 +1,293 @@
+classdef layersClassRealSLD < handle
+    
+    % This is the class definition for the layers block with no absorption.
+
+    properties
+        layersTable = table
+    end
+    
+    properties (Access = private)
+        layersCount
+        layersAutoNameCounter
+        allowedHydration = {'bulk in','bulk out','none'}; 
+    end
+    
+    methods
+        
+        function obj = layersClassRealSLD()
+            % Constructor a layers class including an empty layers table
+            %
+            % layers = layersClassRealSLD();          
+            sz = [0 6];
+            varTypes = {'string','string','string','string','string','string'};
+            varNames = {'Name','Thickness','SLD','Roughness','Hydration','Hydrate with'};
+            obj.layersTable = table('Size',sz,'VariableTypes',varTypes,'VariableNames',varNames);
+            %obj.layersTable(1,:) = {'Layer 1','', '', '', '','bulk out'};
+            obj.layersCount = 0;
+            obj.layersAutoNameCounter = 1;
+            
+        end
+        
+        function obj = addLayer(obj, whatToAdd, paramNames)
```

Comment:
See previous comment, could be better to flip the argument order and use a varargin for whatToAdd

Reply:
Done, and allows removal of some now unnecessary code from projectClass